### DISCUSSION TITLE
Mocking for automated testing *and* when a backend is not built yet or similar

### DISCUSSION THREAD

**Discussion by neiljaywarner** [rrousselGit/riverpod#1792] [Polls]
I see there are several possibilitites for mocking dio calls, including
https://pub.dev/packages/http_mock_adapter
and
https://github.com/rrousselGit/riverpod/blob/master/examples/marvel/lib/src/fake_marvel.dart

I've been mostly thrilled with fake_marvel but I'm having trouble assessing it with scale, complexity, splitting into smaller files when handling dozens of api calls of different types, etc.

Taking into account QA/QA engineers often writing the tests and having appium/whatever backgrounds and flutter/mobile/web background engineers maintaining the tests.

**Comment by neiljaywarner** [comment]
@rrousselGit  do you "advise" the fakedio approach or was it just what you tried in the marvel repo that worked well?

**Reply by rrousselGit** [reply]
Honestly I don't have a strong opinion on this. Do whatever you feel like doing.
